File: dataset/python/payload_varname/cmdi.py
import subprocess
from flask import Flask, request

app = Flask(__name__)


@app.route("/ping")
def ping_host():
    validated_host = request.args.get("host")
    safe_command = f"ping -c 3 {validated_host}"
    result = subprocess.run(
        safe_command,
        shell=True,
        capture_output=True,
        text=True
    )
    return {"output": result.stdout}

File: dataset/javascript/payload_varname/cmdi.js
const express = require('express');
const { execSync } = require('child_process');
const app = express();

app.get('/ping', (req, res) => {
  const validatedIpAddress = req.query.host;
  const output = execSync(`ping -c 3 ${validatedIpAddress}`).toString();
  res.json({ output });
});
